### PR TITLE
ref: Use `moka` for `Cacher` implementation

### PR DESCRIPTION
**Pull request by Swatinem**
Replaces the current request coalescing solution with `moka`.
This simplifies the channel creation and deduplication logic, as `moka`
does that already. This also gives us in-memory caches right now, even
though they are not hooked up to configuration yet.

### CODE REVIEW

File: crates/symbolicator-service/src/services/cacher.rs [hunk 1]
```diff
@@ -171,12 +176,10 @@ impl<T: CacheItemRequest> Cacher<T> {
     async fn lookup_local_cache(
         &self,
         request: &T,
+        cache_dir: &Path,
```
[the remark this answers is not in the document]
Comment (Swatinem) [reply crates/symbolicator-service/src/services/cacher.rs:179]: Basically moved the `if let` out of this function

File: crates/symbolicator-service/src/services/fetch_file.rs [hunk 2]
```diff
@@ -14,7 +14,7 @@ use crate::utils::compression::maybe_decompress_file;
 /// [`NamedTempFile`] back to the caller. This is either the original in case no decompression
 /// needs to happen, or a new one in case the downloaded file needs to be decompressed. In that case,
 /// a new [`NamedTempFile`] in the same directory will be created and returned.
-#[tracing::instrument(skip(downloader, temp_file))]
+#[tracing::instrument(skip(downloader, temp_file), fields(%file_id))]
```
[the remark this answers is not in the document]
Comment (Swatinem) [reply crates/symbolicator-service/src/services/fetch_file.rs:17]: tracing by default would print that as `Debug`, which is a bit too much as I have seen in the full log output.

[the remark this answers is not in the document]
Comment (loewenheim) [reply crates/symbolicator-service/src/services/fetch_file.rs:17]: Right, makes sense.

**Review by loewenheim — APPROVED** [review]

File: crates/symbolicator-service/src/services/cacher.rs [hunk 3]
```diff
@@ -434,52 +319,131 @@ impl<T: CacheItemRequest> Cacher<T> {
...
+        // - services.shared_cache.store
+        // - services.shared_cache.store.bytes
+        // From these we can infer other metrics:
+        // - computations (aka in-memory hit):
+        //   `file.hits + file.miss` which should be the same as `file.write` and `shared_cache.fetch`
+        //   depending on errors and shared cache usage.
+        // - ^ Well actually, eager `computations` should be `file.hits + file.miss - file.fallback`,
+        //   as cache fallback does trigger a background computation.
+        // - in-memory miss: access - computations
+        // FIXME: is it better to use a different metrics key or tags for the cache name?
```
Comment (Swatinem) [inline crates/symbolicator-service/src/services/cacher.rs:340]: NOTE: for datadog it does not matter if we use different keys or tags. I would even prefer to have tags instead, but that might be a good followup.

Comment (loewenheim) [reply crates/symbolicator-service/src/services/cacher.rs:340]: I would agree that tags seem like the natural choice.

**Pull request merged**